### DISCUSSION TITLE
Component parameters cannot be updated when adding a card using Hash route switching

### DISCUSSION THREAD

**Discussion by Guanzong-Wu** [h2oai/wave#1912] [Q&A] [answered]
I'm having trouble trying to customise the plot using components such as sliders, comboboxes, etc., 
Without using hash route switching, each change is saved and reacts instantly. However, when I redesign the UI of my project and add it to a framework that uses hash route switching, every time I try to update the component values, it doesn't save them, even sliderh and combobox don't work, what's wrong? Is it because of the addition of async?

**Comment by mturoci** [comment, marked as answer]
Hello @Kiwi-Rionald. Can you provide a minimal code so that I can get an idea of what you are doing and what might be a problem? Thanks!

**Reply by Guanzong-Wu** [reply]
Sorry for not being clear the first time I asked a question here, and then the problem has been solved because I removed the page that the function would create in the async function after `on()`. It looks like this:

```python
on()
async def func(q: Q).
    del q.page['pageName']
    q.page['pageName'] = ui.form_card()
```

Because I did not understand the mechanism of the update, this will cause me to update the page in the formation of options will disappear, the truth is because this will be written in a kind of "dead loop".
Also, thank you for replying after I closed the question, I'm touched by the nice atmosphere here, thanks again!

**Reply by mturoci** [reply]
Glad you figured it out! Sorry for not reaching out sooner, but you know, the weekend stuff :)
